File: src/main/java/ua/borovyk/catalogue/data/entity/Product.java
package ua.borovyk.catalogue.data.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@ToString
@Entity
@Table(
        name = "product",
        uniqueConstraints = {
                @UniqueConstraint(name = "product_article_ukey", columnNames = {"article"})
        }
)
public class Product {

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private Integer article;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private Double height;

    @Column(nullable = false)
    private Double width;

    @Column(nullable = false)
    private Double deep;

    @ManyToOne(optional = false)
    @JoinColumn(name = "type_id", nullable = false)
    private Type type;

}

File: src/main/ui/src/models/Product.js
import { Model } from 'vue-mc'
import ProductImage from "./ProductImage";

export default class Product extends Model {

    defaults () {
        return {
            id: null,
            article: null,
            name: null,
            description: null,
            price: null,
            image: null,
            height: null,
            width: null,
            depth: null,
            typeId: null,
        }
    }

    mutations () {
        return {
            id: (id) => Number(id) || null,
            article: (article) => Number(article) || null,
            name: (name) => String(name) || null,
            description: (description) => String(description) || null,
            price: (price) => Number(price) || null,
            image: (image) => new ProductImage(image) || null,
            height: (height) => Number(height) || null,
            width: (width) => Number(width) || null,
            depth: (depth) => Number(depth) || null,
            typeId: (typeId) => Number(typeId) || null,
        }
    }
}
